youtube_dl/extractor/infomaniak.py

```diff
@@ -0,0 +1,50 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+from .common import InfoExtractor
+import json
+
+
+class InfomaniakVOD2IE(InfoExtractor):
+    _VALID_URL = r'https?://player\.vod2\.infomaniak\.com/embed/(?P<id>[^/?#&]+)'
+    _TEST = {
+        'url': 'https://player.vod2.infomaniak.com/embed/1jhvl2uq7kr4y',
+        'md5': 'd06fb3fc5a8d7cb4d6e4a0f4e7c5a76a',
+        'info_dict': {
+            'id': '1jhvl2uq7kr4y',
+            'ext': 'mp4',
+            'title': 'RolandCarey2016-05-04.mp4',
+            'thumbnail': 'https://res.vod2.infomaniak.com/1/vod/thumbnail/1jhvl2uq8yqqv.jpg',
+            'duration': 221,
+        }
+    }
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        # no useful data in given url, formatted url below reveals json data
+        data_url = "https://res.vod2.infomaniak.com/{}/vod/share/{}".format(video_id[0], video_id)
+        webpage = self._download_webpage(data_url, video_id)
+
+        data = json.loads(webpage)['data']['media'][0]
+
+        url = data['source']['url']
+        title = data['title']
+        thumbnail = data.get('image').get('url')
+        duration = data.get('duration')
+
+        video_mimetype = data.get('source').get('mimetype')
```

Comment:
Use `traverse_obj()` utility and condition values (imports from `..utils`):
```suggestion
        thumbnail = traverse_obj(data, ('image', 'url'), expected_type=url_or_none)
        duration = traverse_obj(data, 'duration', expected_type=int_or_none)

        video_mimetype = traverse_obj(data, ('source', 'mimetype'), expected_type=lambda x: x.strip() or None)
```